gnucash/register/ledger-core/gnc-ledger-display.c

```diff
@@ -903,5 +903,6 @@ gnc_ledger_display_close (GNCLedgerDisplay *ld)
     if (!ld)
         return;
 
-    gnc_close_gui_component (ld->component_id);
+    if (ld->component_id)
+        gnc_close_gui_component (ld->component_id);
```

Comment:
Won't this leave the register open with a zombie account?

Reply:
I checked that the close_handler in gnc_ledger_display was being called before the gnc_ledger_display_close which unregistered the ld->component_id before it frees the ld so thought this would be OK.

Reply:
What?! The close_handler is supposed to be called by gnc_close_gui_component. It shouldn't get run first.

Regardless, since the close handler doesn't then call component_id = 0 (or better compnent_id = NO_COMPONENT, but you'd have to change this test accordingly) the test won't change anything.

Reply:
Will need to have another look at this and check the order of things again.

Reply:
I think I know what is wrong with this, when the register is registered with the component manager you end up with this setup...
Components: id's
  GncPluginPageRegister:	6
  register-single:	                5
  GncPluginPageRegister:	4
  register-single:	                3
  summary-bar:	                2
  plugin-page-acct-tree:	        1

After the account is destroyed, there is a 'gnc_resume_gui_refresh' which then calls 'gnc_gui_refresh_internal' and on line 692 it calls 'find_component_ids_by_class' which is used in order resulting in 'register-single' being unregistered before 'GncPluginPageRegister' and hence 'ld' being freed but the register not knowing this. If I change g_list_prepend to g_list_append in 'find_component_ids_by_class' the list is reversed or use g_list_reverse and so no error.

What I do not know is whether that is a valid fix ?

Reply:
g_list_append is slow. It would be better to call g_list_reverse and use the reversed list.
We also need ```ld = NO_COMPONENT``` after it has been freed.